$NetBSD: patch-ad,v 1.1 2007/10/10 22:47:46 rillig Exp $

Using macros for types belongs to the bad habits of C programmers.

--- utils/general.h.orig	2002-08-08 13:13:01.000000000 +0200
+++ utils/general.h	2007-10-11 00:45:50.000000000 +0200
@@ -9,9 +9,7 @@
 #ifndef __GENERAL_H__
 #define __GENERAL_H__
 
-#ifndef NULL
-#define NULL 0
-#endif
+#include <stddef.h>
 
 #ifndef FALSE
 #define FALSE 0
@@ -22,7 +20,8 @@
 #endif
 
 #ifndef uint
-#define uint unsigned int
+typedef unsigned int resizeimage_uint;
+#define uint resizeimage_uint
 #endif
 
 #ifndef int8
@@ -46,7 +45,9 @@
 #endif
 
 #ifndef boolean
-#define boolean int
+typedef int boolean;
+#define boolean boolean
+#define HAVE_BOOLEAN
 #endif
 
 #ifndef pointer
